src/utils/daemon.js

```diff
@@ -139,10 +142,82 @@ function checkCorsConfig (ipfsd) {
   store.set('checkedCorsConfig', true)
 }
 
+async function checkIfAddrIsDaemon (addr) {
+  const options = {
+    method: 'GET',
+    host: addr.address,
+    port: addr.port,
+    path: '/api/v0/refs?arg=/ipfs/QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn'
+  }
+
+  return new Promise(resolve => {
+    var req = http.request(options, function (r) {
+      resolve(r.statusCode === 200)
+    })
+
+    req.on('error', () => {
+      resolve(false)
+    })
+
+    req.end()
+  })
+}
+
+const parseCfgMultiaddr = (addr) => (addr.includes('/http')
+  ? multiaddr(addr)
+  : multiaddr(addr).encapsulate('/http')
+)
+
+/*
+const getCurrentApiAddr = async (ipfsd) => {
+  const path = join(ipfsd.repoPath, 'api')
+
+  if (!await fs.pathExists(path)) {
+    return null
+  }
+
+  return fs.readFileSync(path).toString()
+}
+*/
+
+async function checkPorts (ipfsd) {
+  const config = readConfigFile(ipfsd)
+
+  const configApiMa = parseCfgMultiaddr(config.Addresses.API)
+  const configGatewayMa = parseCfgMultiaddr(config.Addresses.Gateway)
+
+  const isApiMaDaemon = await checkIfAddrIsDaemon(configApiMa.nodeAddress())
+  const isGatewayMaDaemon = await checkIfAddrIsDaemon(configGatewayMa.nodeAddress())
+
+  if (isApiMaDaemon && isGatewayMaDaemon) {
+    // TODO: BEING USED BY A DAEMON
+    return
+  }
+
+  const apiPort = parseInt(configApiMa.nodeAddress().port)
```

Comment:
JS uses [fancy heuristic to calculate radix based on input](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Description), it is usually safer to provide explicit one:
```suggestion
  const apiPort = parseInt(configApiMa.nodeAddress().port, 10)
```